robottelo/hosts.py

```diff
@@ -226,16 +226,33 @@ def satellite(self):
             self._satellite = Satellite()
         return self._satellite
 
+    @property
+    def _sat_host_record(self):
+        """Provide access to this host's Host record if it exists."""
+        host = None
+        with contextlib.suppress(IndexError):
+            host = self.satellite.api.Host().search(query={'search': self.hostname})[0]
+        return host
+
+    def _delete_host_record(self):
+        """Delete the Host record of this host from Satellite."""
+        if h_record := self._sat_host_record:
+            h_record.delete()
+        else:
+            logger.warning(
+                f'Attempted to delete Host {self.hostname}, but it was not found in Satellite'
+            )
+
     @property
```

Comment:
I don't think it makes sense to write new properties for same thing, instead shouldn't we cover the other posiblity of host is registered to capsule or to server other than CDN into the nailgun_host property itself, wdyt?

Reply:
The main issue is that `subscription-manager identity` returns nothing on stdout, therefore the return value of `self.identity` is `None`. Therefore `self.identity.get('registered_to') == self.satellite.hostname` is evaluated as `False` and `nailgun_host` then returns `None`.
I have tried to avoid extending the logic of the nailgun host to keep it simple. Therefore `type(self) is not Satellite and self.nailgun_host` evaluates to `True and False` --> `False` in `teardown`, so the host entity does not get deleted.

I am open to changing it. Feel free to share your ideas around this.

Reply:
yes, `subscription-manager identity` returns some uuid and org details in stdout, and identity property returns a dict which also contains the `registered_to: subscription_config['server']['hostname']`, so I think there'll be a some value present always but not sure about self.satellite.hostname, while I don't see any issue with removing nailgun_host from `type(self) is not Satellite and self.nailgun_host` it's fine, but I still feel same nailgun_host property should be fixed for this issue and not in other similar properties